File: khalti-android/src/main/java/khalti/checkOut/EBanking/deepLinkReceiver/DeepLinkPresenter.java
package khalti.checkOut.EBanking.deepLinkReceiver;

import android.support.annotation.NonNull;

import java.util.HashMap;

import khalti.checkOut.api.Config;
import khalti.checkOut.api.OnCheckOutListener;
import khalti.rxBus.RxBus;
import khalti.utils.EmptyUtil;
import khalti.utils.GuavaUtil;
import khalti.utils.LogUtil;
import khalti.utils.Store;

class DeepLinkPresenter implements DeepLinkContract.Listener {
    @NonNull
    private final DeepLinkContract.View mDeepLinkView;

    DeepLinkPresenter(@NonNull DeepLinkContract.View mDeepLinkView) {
        this.mDeepLinkView = GuavaUtil.checkNotNull(mDeepLinkView);
        mDeepLinkView.setListener(this);
    }

    @Override
    public void receiveEBankingData() {
        HashMap<String, Object> map = mDeepLinkView.receiveEBankingData();
        Config config = EmptyUtil.isNotNull(Store.getConfig()) ? Store.getConfig() : mDeepLinkView.getConfigFromFile();

        if (EmptyUtil.isNotNull(map) && EmptyUtil.isNotNull(config)) {
            OnCheckOutListener onCheckOutListener = config.getOnCheckOutListener();
            onCheckOutListener.onSuccess(map);
        }
        mDeepLinkView.closeDeepLink();
        RxBus.getInstance().post("close_check_out", null);
    }
}

File: khalti-android/src/main/java/com/khalti/checkout/banking/deepLinkReceiver/DeepLinkPresenter.kt
package com.khalti.checkout.banking.deepLinkReceiver

import com.khalti.checkout.helper.Config
import com.khalti.utils.EmptyUtil
import com.khalti.utils.GuavaUtil
import com.khalti.utils.Store

internal class DeepLinkPresenter(view: DeepLinkContract.View) : DeepLinkContract.Presenter {

    private val view: DeepLinkContract.View = GuavaUtil.checkNotNull(view)

    init {
        view.setPresenter(this)
    }

    override fun onCreate() {
        val map = view.receiveEBankingData()
        val config: Config? = Store.getConfig()

        if (EmptyUtil.isNotNull(map) && EmptyUtil.isNotNull(config)) {
            val onCheckOutListener = config!!.onCheckOutListener
            if (EmptyUtil.isNotNull(onCheckOutListener)) {
                onCheckOutListener.onSuccess(map!!)
            }
        }
        view.closeDeepLink()
        Store.getCheckoutEventListener().closeCheckout()
    }

    override fun onDestroy() {
    }
}